cli/sub_command.go

```diff
@@ -317,47 +316,55 @@ func (c *subCmd) validateInputs(nc *nats.Conn, mgr *jsm.Manager) error {
 	}
 
 	if c.jetStream {
+		js, err := jetstream.New(nc)
+		if err != nil {
+			return err
+		}
 		// if we are passed subjects without a stream, we determine the stream name before trying to load it for further checks
 		if c.stream == "" {
-			if len(c.subjects) > 1 {
-				return fmt.Errorf("cannot enable direct gets: cannot perform direct gets from multiple subjects")
-
+			c.stream, err = js.StreamNameBySubject(ctx, c.subjects[0])
+			if errors.Is(err, jetstream.ErrStreamNotFound) {
+				return fmt.Errorf("no stream found with subject: %s", c.subjects[0])
 			}
-			streams, err := mgr.StreamNames(&jsm.StreamNamesFilter{
-				Subject: c.subjects[0],
-			})
 			if err != nil {
 				return err
 			}
-
-			if len(streams) > 1 {
-				return fmt.Errorf("cannot enable direct gets: subject mapped to multiple streams")
-			} else if len(streams) == 0 {
-				return fmt.Errorf("cannot enable direct gets: subject mapped to no streams")
-			}
-
-			c.stream = streams[0]
 		}
 
-		var err error
-		c.streamObj, err = mgr.LoadStream(c.stream)
+		c.streamObj, err = js.Stream(ctx, c.stream)
 		if err != nil {
 			return err
 		}
 
 		// Check if the stream is a Work Queue
-		c.direct = c.direct || c.streamObj.Retention() == api.RetentionPolicy(nats.WorkQueuePolicy)
+		c.direct = c.direct || c.streamObj.CachedInfo().Config.Retention == jetstream.WorkQueuePolicy
 
 		if c.direct {
+			if len(c.subjects) > 1 {
+				return fmt.Errorf("cannot enable direct gets: cannot perform direct gets from multiple subjects")
+
+			}
 			// Direct Addressing needs to be allowed on the stream
-			if !c.streamObj.DirectAllowed() {
+			if !c.streamObj.CachedInfo().Config.AllowDirect {
 				return fmt.Errorf("cannot enable direct gets: allow_direct is not set to true for stream %s", c.stream)
 			}
 			err := iu.RequireAPILevel(mgr, 1, "subscribing using direct get requires NATS Server 2.11")
 			if err != nil {
 				return err
 			}
 		}
+
```

Comment:
I'm not sure about these. We could probably add at least some of this functionality.

Reply:
about needing 2.11? I am ok with that, good things come to those who upgrade